src/worker/utils/chem.ts

```diff
@@ -136,6 +136,99 @@ export const getMatchingSubstructure = ({ structure, substructure }: { structure
   return { matchingAtoms: atoms, matchingBonds: bonds };
 };
 
+export const isValidMolBlock = (mdl: string) => {
+  if (!mdl.includes('M  END')) return false;
+  const mol = get_molecule(mdl, globalThis.workerRDKit);
+  if (!mol) return false;
+  try {
+    return mol.is_valid();
+  } finally {
+    release_molecule(mol);
+  }
+};
+
+/**
+ * Convert molecule structure to given notation (smiles, smarts, molblock, ...)
+ * TODO some notations like inchi don't work with qmol
+ */
+export const convertMolNotation = ({
+  moleculeString,
+  targetNotation,
+  sourceNotation,
+  useQMol = false,
+}: {
+  moleculeString: string;
+  targetNotation: MolNotation;
+  sourceNotation?: SourceMolNotation;
+  useQMol?: boolean;
+}): string | null => {
+  if (sourceNotation != null) {
+    if (sourceNotation === targetNotation) throw new Error('Source and target notations must differ');
+    if (!_validateSource(moleculeString, sourceNotation)) throw new Error('Molecule string not valid');
+  }
+  const mol = useQMol
+    ? globalThis.workerRDKit.get_qmol(moleculeString)
+    : get_molecule(moleculeString, globalThis.workerRDKit);
+  if (!mol) return null;
+  try {
+    return mol[`get_${targetNotation}`]();
+  } catch (e) {
+    console.error(e);
+    throw new Error('Target notation not implemented');
+  } finally {
+    useQMol ? mol?.delete() : release_molecule(mol);
+  }
+};
+
+export const getNewCoords = (structure: string, useCoordGen?: boolean) => {
+  const mol = get_molecule(structure, globalThis.workerRDKit);
+  if (!mol) return null;
+  try {
+    const mdl = useCoordGen != null ? mol.get_new_coords(useCoordGen) : mol.get_new_coords();
```

Comment:
was this meant to be `useCoordGen !== undefined`? or `export const getNewCoords = (structure: string, useCoordGen?: boolean | null = null) => {` ?

Reply:
`useCoordGen != null` check if `useCoordGen` is not `null` or `undefined`.

Reply:
Yeah I did it like that to avoid passing a null and throw an error, but I just check it and `get_new_coords` seems working fine with `null` and `undefined`.
Changed to `useCoordGen !== undefined`